lib/WorseReflection/Bridge/TolerantParser/Diagnostics/UndefinedVariableProvider.php

```diff
@@ -307,6 +307,22 @@ function preg_match(string $pattern, string $string, array &$matches): bool {}
                 Assert::assertCount(0, $diagnostics);
             }
         );
+        yield new DiagnosticExample(
+            title: "SAPI global variables",
+            source: <<<'PHP'
+                <?php
+                if ($argc === 2) {
+                    echo "Hello ".$argv[1].PHP_EOL;
+                } else {
+                    echo "Usage ".__FILE__. " <name>";
+                }
+            PHP,
+            valid: true,
+            assertion: function (Diagnostics $diagnostics): void {
+                Assert::assertCount(0, $diagnostics);
+            }
```

Comment:
This assertion is kinda superfluous as it is also checked when `valid: true` is set. Maybe we could make the assertion optional when `valid` is true?

Reply:
yeah possibly